### DISCUSSION TITLE
Waste Controling

### DISCUSSION THREAD

**Discussion by domgggg** [MikaylaFischler/cc-mek-scada#444] [Support] [answered]
In the video you made about connecting wires to emergency coolant and waste processing. You connect the wires directly to the rtu, but i want to know how you can do that from a distance. like with multiple connections.

![2024-03-06_16 14 36](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/b508c57b-acdf-452c-a72a-8f7918b25679)

**Comment by MikaylaFischler** [comment]
A single wire can carry up to 16 signals by using the bundled/colored redstone options. You can send a wire across as many connectors as you want for how ever far it needs to go. You just need an interface connector on the RTU and then connectors at the other end configured properly for their color and input/output mode.

Edit: If it spans chunks and those chunks aren't loaded while others are, my assumption is that it wouldn't work, but I haven't tested that.

**Reply by domgggg** [reply]
Should i make the connectors the corresponding color to the connectors at the end connecting to the pipe? Also should I make them input or output.

**Reply by MikaylaFischler** [reply]
That should be covered in the video. The only connector colors that matter are the ones touching the things you want to use as inputs/outputs. The ones in between can be whatever color you like. The ones touching the pipes need to be set as outputs, and each should have it's own color that you will additionally set in the RTU configurator, per the video. Docs on redstone configuration in the RTU are here: https://github.com/MikaylaFischler/cc-mek-scada/wiki/RTU-Gateway-Configurator#redstone-connections

**Reply by domgggg** [reply]
okay thanks

**Reply by domgggg** [reply]
![2024-03-08_15 24 28](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/d87cce2e-5af6-4944-9af7-ab4cbe0689fa)
The Redstone is apparently not connected.

**Reply by MikaylaFischler** [reply]
RT is the co-routine status. Redstone I/O doesn't have a co-routine since it doesn't need to interface with Mekanism devices, meaning I can immediately read/write rather than wait for Mekanism to get back to me

**Reply by MikaylaFischler** [reply]
So to clarify, the light next to the RTU name (REDSTONE, SPS, etc) is its status (if it is actually connected and functioning) and the one next to RT is the co-routine status, if applicable. The co-routine status should only be off if it is redstone, the whole program is exiting, or there was a bug that crashed that specific co-routine.

**Reply by domgggg** [reply]
But everytime I try to choose to make anti matter or any other one it just doesnt shut any valves

**Reply by MikaylaFischler** [reply]
Are your pipes redstone sensitive? As in right clicked with a Mekanism configurator tool?

**Reply by domgggg** [reply]
yes they are I just checked.

**Reply by MikaylaFischler** [reply]
Does the flow monitor show them changing states, and does it show them as connected?

**Reply by domgggg** [reply]
how would i check

**Reply by MikaylaFischler** [reply]
You could send a screenshot of the valve and flow monitor (see the titles of the monitors)

**Reply by domgggg** [reply]
![image_2024-03-11_175802042](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/7a57f211-afbb-434a-919c-8dea39791d46)

![image_2024-03-11_175852188](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/a1e66813-f7cc-4ba1-8696-b4dce97e62fe)
also this is how its connected.

**Reply by MikaylaFischler** [reply]
That shows that you're missing your polonium and polonium pellets valve connections, and that the antimatter one is open and the plutonium one is closed. Can you show the connections to the pipes?

**Reply by domgggg** [reply]
![image_2024-03-11_180412242](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/6b20d97e-ee33-4e33-85b6-5731ef299a27)
The colors are default im pretty sure.

**Reply by domgggg** [reply]
I only want to have two

**Reply by MikaylaFischler** [reply]
The lime one isn't in the right place for one, and you need to have all 4 waste valves if you aren't only using plutonium. I do not see why they wouldn't be working if those specific segments of pipe are set as redstone sensitive and both of those connectors are set as outputs and the screen shows them as connected and set. I'm not sure which part of the setup you have incorrect based off of these images. You could show your RTU redstone configuration as well where it lists them in the configurator.

You also only need to run one total redstone wire in your example.

https://github.com/MikaylaFischler/cc-mek-scada/wiki/Waste-Setup

**Reply by MikaylaFischler** [reply]
Correction, you don't really need any if you only have a single route for waste. If you have plutonium and polonium you need WASTE_PU and WASTE_PO. If you have all three waste products, you need all four valves.

**Reply by domgggg** [reply]
I put down all four valves. In the rtu config all four ports should be their and are to reactor unit 1.

![image_2024-03-11_184829260](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/49549c6e-9131-4192-a09b-cae89beffe0a)

![Screenshot 2024-03-11 184805](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/853b3a38-62fe-4e52-b005-9c1e72ad520d)

![Screenshot 2024-03-11 184745](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/c52333e5-d990-4217-8294-59d155f221ac)

![Screenshot 2024-03-11 184716](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/79e221c6-9e0d-4506-a357-273e27302581)

![Screenshot 2024-03-11 184700](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/49c56bb9-6fa7-4456-80c9-b159620428c3)

**Reply by MikaylaFischler** [reply]
Can you try, closer to the RTU, connecting the four outputs to redstone lamps and see if they light up?

**Reply by domgggg** [reply]
I cant do closer to the rtu but I will if its the last resort. But yea ill connect it to redstone lamps

**Reply by domgggg** [reply]
it didnt work non of the lights turned on.

**Reply by MikaylaFischler** [reply]
Is the interface connector on the "right" side of the computer? That would mean you put the wired modem on the front, which is non-standard but technically works. I can't tell from the one picture containing the computer that you sent.

**Reply by domgggg** [reply]
![image_2024-03-17_114857837](https://github.com/MikaylaFischler/cc-mek-scada/assets/160686612/ed65451a-b582-43a8-99dd-4c59aed3b9aa)
 thats my rtu

**Reply by MikaylaFischler** [reply, marked as answer]
Well that's the issue then. The configuration you sent said the redstone is on the right, but it's on the left in that picture. You need to change the configuration entries to say left or move the interface connector to the right

**Reply by domgggg** [reply]
okay thanks

**Reply by domgggg** [reply]
Its not working still the llight on the rtu isnt on and we have the same problems.

**Reply by MikaylaFischler** [reply]
What light on the RTU? If it's the RT light it isn't supposed to be on for the redstone entries. Did you change all the entries to left or did you move it?

**Reply by MikaylaFischler** [reply]
For faster back-and-forth, if you have a Discord, you can join the one for this project and start a thread in #support-forum. If not, then we can continue here.

If you can share a screenshot of your RTU configuration, your RTU screen when its running, the RTU in-world, the full redstone wire run, the flow monitor, the main monitor, and the unit monitor, then maybe I can catch something I haven't already. Please also state what you are trying to do that is "not working" and results in "the same problems".

**Reply by domgggg** [reply]
never mind all good. thanks
